orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltOrchestrator.java

```diff
@@ -258,8 +261,11 @@ public void bootstrap(List<GatewayConfig> allGatewayConfigs, Set<Node> targets,
         }
     }
 
-    private void initializePillar(Set<Node> allNodes, ExitCriteriaModel exitModel, Set<String> gatewayTargetIpAddresses, SaltConnector sc) throws Exception {
-        OrchestratorBootstrap hostSave = new PillarSave(sc, gatewayTargetIpAddresses, allNodes);
+    private void saveHostsPillar(OrchestratorAware stack, ExitCriteriaModel exitModel,
+            Set<String> gatewayTargetIpAddresses, SaltConnector sc) throws Exception {
+        Set<? extends OrchestrationNode> allNodesForOrchestration = stack.getAllNodesForOrchestration();
+        Set<Node> allNodes = allNodesForOrchestration.stream().map(OrchestrationNode::getNode).collect(Collectors.toSet());
```

Comment:
this stream could be a default implementation in the `OrchestratorAware` interface as  see it's used in multiple places